Bookish/Bookish.Web/Controllers/HomeController.cs

```diff
@@ -56,13 +63,22 @@ public IActionResult AddBookPost(string title, string authors, string isbn, int
         {
             var isbnError = libraryService.BookExists(isbn);
             var copyError = copies < 1;
-            if (isbnError || copyError)
+            var invalidFormat = isbn?.Replace("-", "").Length != 13;
+            if (isbnError || copyError || invalidFormat)
             {
                 return RedirectToAction("AddBook", new { isbn, title, authors, copies });
             }
             
             libraryService.AddBook(isbn, title, authors, copies);
-            return RedirectToAction("Catalogue");
+            return RedirectToAction("AddedBook", new { isbn, copies });
+        }
+
+        public IActionResult AddedBook(string isbn, int copies)
+        {
+            var title = libraryService.GetBook(isbn).title;
```

Comment:
This should happen in barcodeService. The controller should only need to call one service method 🙂